packages/service-backend/src/course/course.service.ts

```diff
@@ -315,4 +354,51 @@ export class CourseService {
 
     return { code, expiresAt: expirationDate };
   }
+
+  static async UpdateCourseCompletionFromLesson(
+    sectionId: string,
+    userId: string,
+  ) {
+    const courseId = await prisma.section.findUnique({
+      where: { id: sectionId },
+      select: {
+        course_id: true,
+      },
+    });
+
+    const courseSectionsIds = (
+      await prisma.section.findMany({
+        where: {
+          course_id: courseId.course_id,
+        },
+        select: { id: true },
+      })
+    ).map((lesson) => lesson.id);
+
+    const courseUserToSectionStatuses: Status[] = (
+      await prisma.usertoSection.findMany({
+        where: {
+          id: { in: courseSectionsIds },
+        },
+        select: { status: true },
+      })
+    ).map((userToSectionObject) => userToSectionObject.status);
+
+    if (
+      courseUserToSectionStatuses.every(
+        (status) => status === Status.COMPLETED,
+      ) === true &&
+      courseSectionsIds.length === courseUserToSectionStatuses.length
```

Comment:
we can simplify this part the way we did in the other file with one query. see bellow